cmd/group-auto-updater/main.go

```diff
@@ -0,0 +1,143 @@
+// The purpose of this tool is to read a peribolos configuration
+// file, get the admins/members of a given organization and
+// update the users of a specific group in an Openshift cluster.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+
+	userV1 "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+
+	"github.com/openshift/ci-tools/pkg/steps"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	group           string
+	peribolosConfig string
+	org             string
+
+	dryRun            bool
+	determinizeOutput bool
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.group, "group", "", "The group that will be updated in the cluster.")
+	fs.StringVar(&o.peribolosConfig, "peribolos-config", "", "Peribolos configuration file")
+	fs.StringVar(&o.org, "org", "", "Org from peribolos configuration")
+
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Print the generated group without updating it")
+	fs.BoolVar(&o.determinizeOutput, "determinize-output", false, "Determinize dry run's output by ordering the created object and removing the unnecessary fields")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func validateOptions(o options) error {
+	if len(o.group) == 0 {
+		return fmt.Errorf("--group is not specified")
+	}
+	if len(o.peribolosConfig) == 0 {
+		return fmt.Errorf("--peribolos-config is not specified")
+	}
+	if len(o.org) == 0 {
+		return fmt.Errorf("--org is not specified")
+	}
+
+	if !o.dryRun && o.determinizeOutput {
+		return fmt.Errorf("--determinize-output has to specified with --dry-run")
+	}
+
+	return nil
+}
+
+func getUserV1Client() (*userV1.UserV1Client, error) {
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load cluster clusterConfig: %v", err)
+	}
+
+	userV1Client, err := userV1.NewForConfig(clusterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not create user openshift client: %v", err)
+	}
+
+	return userV1Client, nil
+}
+
+func getUsersFromPeribolosConfig(peribolosConfig org.FullConfig, pOrg string) []string {
+	users := sets.NewString()
+	users.Insert(peribolosConfig.Orgs[pOrg].Admins...)
+	users.Insert(peribolosConfig.Orgs[pOrg].Members...)
+	return users.List()
+}
+
+func updateGroup(group string, users []string, groupV1Interface userV1.GroupInterface, dryLogger *steps.DryLogger, dryRun bool) error {
```

Comment:
Similarly, this is called once and doesn't do anything complicated -- lets just inline it into `main()`